core/domain/storage_model_audit_jobs_test.py

```diff
@@ -0,0 +1,102 @@
+    'SolutionHitEventLogEntryModel',
+    'StartExplorationEventLogEntryModel',
+    'MaybeLeaveExplorationEventLogEntryModel',
+    'CompleteExplorationEventLogEntryModel',
+    'RateExplorationEventLogEntryModel',
+    'StateHitEventLogEntryModel',
+    'StateCompleteEventLogEntryModel',
+    'LeaveForRefresherExplorationEventLogEntryModel',
+    'ExplorationStatsModel',
+    'ExplorationIssuesModel',
+    'PlaythroughModel',
+    'LearnerAnswerDetailsModel',
+    'ExplorationAnnotationsModel',
+    'StateAnswersModel',
+    'StateAnswersCalcOutputModel',
+    'StateIdMappingModel'
+}
+
+
+class StorageModelAuditJobsTest(test_utils.GenericTestBase):
+    """Tests for Oppia storage model audit jobs."""
+
+    def test_all_models_have_audit_jobs(self):
+        all_model_names = []
+
+        # As models.NAMES is an enum, it cannot be iterated. So we use the
+        # __dict__ property which can be iterated.
+        for name in models.NAMES.__dict__:
+            if '__' not in name:
+                all_model_names.append(name)
+
+        names_of_ndb_model_classes = []
+        for name in all_model_names:
+            # We skip base models since there are no specific audit jobs
+            # for base models. The audit jobs for subclasses of base models
+            # cover the test cases for base models, so extra audit jobs
+            # for base models are not required.
+            if name == 'base_model':
+                continue
+            (module, ) = models.Registry.import_models([name])
+            for member_name, member_obj in inspect.getmembers(module):
+                if inspect.isclass(member_obj):
+                    clazz = getattr(module, member_name)
+                    if clazz.__name__ in MODEL_CLASS_NAMES_TO_EXCLUDE:
+                        continue
+                    ancestor_names = [
+                        base_class.__name__ for base_class in clazz.__bases__]
+                    if (
+                            'ndb.Model' in ancestor_names or
+                            'BaseModel' in ancestor_names or
+                            'VersionedModel' in ancestor_names):
+                        names_of_ndb_model_classes.append(clazz.__name__)
+
+        names_of_all_audit_job_classes = []
+        for name, clazz in inspect.getmembers(
+                prod_validation_jobs_one_off, predicate=inspect.isclass):
+            ancestor_names = [
+                base_class.__name__ for base_class in clazz.__bases__]
+            if 'ProdValidationAuditOneOffJob' in ancestor_names:
+                names_of_all_audit_job_classes.append(name)
+
+        model_class_names_with_missing_audit_jobs = [
+            model_class_name
+            for model_class_name in names_of_ndb_model_classes if (
+                model_class_name + 'AuditOneOffJob' not in (
+                    names_of_all_audit_job_classes))]
+        if model_class_names_with_missing_audit_jobs:
```

Comment:
Probably should do self.assertEqual(...) and add a msg argument, following the usual unittest assertion framework.

Reply:
Done in #7101